contrib/neml2.mk

```diff
@@ -1,7 +1,59 @@
-ifeq ($(ENABLE_LIBTORCH),true)
-NEML2_DIR            ?= $(BLACKBEAR_DIR)/contrib/neml2
+# There are 2 influential conf vars.
+# They are controlled by the MOOSE configure script.
+#
+# ENABLE_LIBTORCH: True if we should compile with libTorch AND an installation
+#                  of libTorch is available. False otherwise.
+# LIBTORCH_DIR:    The path to the libTorch installation. NEML2 will use this
+#                  libTorch as the backend.
+#
+# There are 2 influential env vars.
+#
+# ENABLE_NEML2:  If true, BlackBear will attempt to compile with NEML2 at
+#                the NEML2_DIR. If false, BlackBear will not be compiled with
+#                NEML2 (even if the NEML2 submodule is init'ed or a valid
+#                NEML2_DIR is supplied). This variable defaults to true.
+# NEML2_DIR:     The path to a valid NEML2 checkout. This variable defaults to
+#                the NEML2 submodule. The user is responsible for setting this
+#                variable if a custom NEML2 should be used. The compilation
+#                will terminate with an error message if ENABLE_NEML2 is true
+#                AND a NEML2 checkout cannot be found.
 
-ifneq ($(wildcard $(NEML2_DIR)/CMakeLists.txt),)
+ENABLE_NEML2 ?= true
+
+ifneq ($(ENABLE_NEML2),true)
+
+$(info Not compiling BlackBear with NEML2 because ENABLE_NEML2 is set to false.)
+
+else
+
+ifneq ($(ENABLE_LIBTORCH),true)
+
+$(error Attempting to compile Blackbear with NEML2, but libTorch is not enabled. \
+  To enable libTorch, configure MOOSE with the --with-libtorch option. \
+  To disable NEML2, set ENABLE_NEML2 to false)
+
+endif
```

Comment:
@loganharbour so this is the condition that prevents all the BlackBear, Grizzly recipes to fail. You will need to explicitly set `ENABLE_NEML2=false`.

Alternatively, I can change this error to a warning message, and skip compiling NEML2. Though I know you discourage this behavior.

The documentation should build just fine with and without NEML2. The only constraint is that the actual recipe that generates the docs which get synced to the public website should be built with NEML2 as well as libTorch.